psl-core/src/main/java/org/linqs/psl/application/learning/weight/search/grid/BaseGridSearch.java

```diff
@@ -94,6 +114,23 @@ protected void doLearn() {
 
             // Set the weights for the current round.
             getWeights(weights);
+
+            // Check if we have explored this configuration before.
+            System.arraycopy(weights, 0, unitWeightVector, 0, weights.length);
+            MathUtils.toUnit(unitWeightVector);
+
+            // Round each weight to 5 decimal places.
+            for(int i = 0; i < unitWeightVector.length; i ++){
+                unitWeightVector[i] = (double)Math.round(unitWeightVector[i] * 100000d) / 100000d;
```

Comment:
When using floating point constants, always use a decimal point (even if it is just a ".0").
(So, `100000.0` instead of `100000d`.)